File: traversal/dfs/python/DFS.py
class Graph:

    def __init__(self, vertices):
        self.V = vertices
        self.graph = [[] for i in range(self.V)]        # adjacency list representation
        self.visited = [False for i in range(self.V)]

    def add_edge(self, src, dest):
        self.graph[src].append(dest)

    def dfs_traversal(self, v):
        self.visited[v] = True
        print(v, end=' ')

        # recur for all adjacent vertices
        for adj in self.graph[v]:
            if not self.visited[adj]:
                self.dfs_traversal(adj)

    def dfs_iterative(self, s):
        self.visited = [False for i in range(self.V)]

        stack = [s]

        while len(stack):
            s = stack.pop()

            if not self.visited[s]:
                print(s, end=' ')
                self.visited[s] = True

            for v in self.graph[s]:
                if not self.visited[v]:
                    stack.append(v)


graph = Graph(13)

graph.add_edge(1, 2)
graph.add_edge(1, 3)
graph.add_edge(1, 5)
graph.add_edge(2, 4)
graph.add_edge(2, 10)
graph.add_edge(2, 5)
graph.add_edge(2, 7)
graph.add_edge(3, 6)
graph.add_edge(4, 7)
graph.add_edge(5, 2)
graph.add_edge(5, 8)
graph.add_edge(6, 9)
graph.add_edge(8, 11)
graph.add_edge(11, 12)

print("DFS Recursive")
graph.dfs_traversal(1)

print("DFS Iterative")
graph.dfs_iterative(1)

File: traversal/dfs/javascript/DFS.js
class Graph{

    constructor(noOfVertices){
        this.V = noOfVertices
        this.edges = []
        this.visited = new Array(this.V).fill(false)
        this.result = []

        for(let i=0; i<this.V; i++)
            this.edges.push([]);
    }

    addEdge(src, dest){
        this.edges[src].push(dest);
    }

    dfsTraversal(v){
        this.visited[v] = true
        this.result.push(v)

        for(let i=0; i<this.edges[v].length; i++){
            let adj = this.edges[v][i]
            if(!this.visited[adj])
                this.dfsTraversal(adj);
        }
    }

    dfsIterative(s){
        this.visited.fill(false);
        this.result = []
        let stack = [s]

        while(stack.length!=0){
            s = stack.pop()

            if(!this.visited[s]){
                this.result.push(s)
                this.visited[s] = true
            }

            for(let i=0; i<this.edges[s].length; i++){
                let adj = this.edges[s][i]
                if(!this.visited[adj])
                    stack.push(adj);
            }
        }
        console.log(this.result);
    }

}

var graph = new Graph(12)

graph.addEdge(0, 1)
graph.addEdge(0, 2)
graph.addEdge(0, 4)
graph.addEdge(1, 3)
graph.addEdge(1, 9)
graph.addEdge(1, 4)
graph.addEdge(1, 6)
graph.addEdge(2, 5)
graph.addEdge(3, 6)
graph.addEdge(4, 1)
graph.addEdge(4, 7)
graph.addEdge(5, 8)
graph.addEdge(7, 10)
graph.addEdge(10, 11)

graph.dfsTraversal(0);
console.log(graph.result);

graph.dfsIterative(0)
